demo/src/main/java/com/orange/ods/demo/ui/components/textfields/TextFieldCustomizationState.kt

```diff
@@ -15,41 +15,141 @@ import androidx.compose.runtime.MutableState
-    trailingElement: MutableState<TrailingElement> = rememberSaveable { mutableStateOf(TrailingElement.NONE) }
+    value: MutableState<String> = rememberSaveable { mutableStateOf("") },
+    inputType: MutableState<InputType> = rememberSaveable { mutableStateOf(InputType.SingleLine) },
+    leadingIcon: MutableState<Boolean> = rememberSaveable { mutableStateOf(false) },
+    displayType: MutableState<DisplayType> = rememberSaveable { mutableStateOf(DisplayType.Default) },
+    trailingElement: MutableState<TrailingElement> = rememberSaveable { mutableStateOf(TrailingElement.None) },
+    visualisationIcon: MutableState<Boolean> = rememberSaveable { mutableStateOf(true) },
+    characterCounter: MutableState<Boolean> = rememberSaveable { mutableStateOf(false) },
+    softKeyboardType: MutableState<SoftKeyboardType> = rememberSaveable { mutableStateOf(SoftKeyboardType.Text) },
+    softKeyboardAction: MutableState<SoftKeyboardAction> = rememberSaveable { mutableStateOf(SoftKeyboardAction.None) },
+    softKeyboardCapitalization: MutableState<Boolean> = rememberSaveable { mutableStateOf(false) }
 ) =
-    remember(leadingIconChecked, displayType, trailingElement) {
-        TextFieldCustomizationState(leadingIconChecked, displayType, trailingElement)
+    remember(
+        value,
+        inputType,
+        leadingIcon,
+        displayType,
+        trailingElement,
+        visualisationIcon,
+        characterCounter,
+        softKeyboardType,
+        softKeyboardAction,
+        softKeyboardCapitalization
+    ) {
+        TextFieldCustomizationState(
+            value,
+            inputType,
+            leadingIcon,
+            displayType,
+            trailingElement,
+            visualisationIcon,
+            characterCounter,
+            softKeyboardType,
+            softKeyboardAction,
+            softKeyboardCapitalization
+        )
     }
 
 class TextFieldCustomizationState(
-    val leadingIconChecked: MutableState<Boolean>,
+    val value: MutableState<String>,
+    val inputType: MutableState<InputType>,
+    val leadingIcon: MutableState<Boolean>,
     val displayType: MutableState<DisplayType>,
-    val trailingElement: MutableState<TrailingElement>
+    val trailingElement: MutableState<TrailingElement>,
+    val visualisationIcon: MutableState<Boolean>,
+    val characterCounter: MutableState<Boolean>,
+    val softKeyboardType: MutableState<SoftKeyboardType>,
+    val softKeyboardAction: MutableState<SoftKeyboardAction>,
+    val softKeyboardCapitalization: MutableState<Boolean>
 ) {
+    companion object {
+        const val TextFieldMaxChars = 20
+    }
+
+    enum class InputType {
+        SingleLine, MultiLine, TextArea
+    }
+
     enum class DisplayType {
-        DEFAULT, ERROR, DISABLED
+        Default, Error, Disabled
     }
 
     enum class TrailingElement {
-        NONE, ICON, TEXT
+        None, Icon, Text
     }
 
+    enum class SoftKeyboardType {
```

Comment:
I modified this enum like this to map strings with keyboard types:
```
    
enum class SoftKeyboardType(val keyboardType: KeyboardType, @StringRes val labelRes: Int) {
        Text(KeyboardType.Text, R.string.component_text_field_keyboard_type_text),
        Decimal(KeyboardType.Decimal, R.string.component_text_field_keyboard_type_decimal),
        Email(KeyboardType.Email, R.string.component_text_field_keyboard_type_email),
        Number(KeyboardType.Number, R.string.component_text_field_keyboard_type_number),
        Phone(KeyboardType.Phone, R.string.component_text_field_keyboard_type_phone),
        Url(KeyboardType.Uri, R.string.component_text_field_keyboard_type_url);
    }
```
Same for `SoftKeyboardAction`.

Do you think this would be better to have a `Map<KeyboardType, @StringRes Int>`?